lib/hedwig_sms.ex

```diff
@@ -8,7 +8,7 @@ defmodule Hedwig.Adapters.SMS do
   @doc false
   def init({robot, opts}) do
     HTTPoison.start
-    Hedwig.Robot.register(robot, opts[:name])
+    :global.register_name(opts[:name], robot)
```

Comment:
Speaking of `handle_connect/1` you probably need to call `Hedwig.Robot.handle_connect(robot)` after you initialize so that the robot's callback will be invoked.